inductiva/projects/project.py

```diff
@@ -55,341 +30,143 @@ def get_projects():
-    of an opened project will be associated with that project and can
-    be retrieved altogether using the project as a filter. A project
-    can be created and managed declaratively or using a context
-    manager.
- 
-    Example usage:
- 
-    .. code-block:: python
 
-        project = inductiva.projects.Project("test_project")
-        project.start()
-        simulator.run(...)  # Submitted to  `'test_project'`
-        project.stop()
- 
-    This is equivalent to:
- 
-    .. code-block:: python
+    Groups related tasks together under a single project.
 
-        with inductiva.projects.Project("test_project"):
-            simulator.run(...)
+    Example:
+        project = inductiva.projects.Project("test_project")
 
+        task_1 = simulator.run(...)
+        project.add_task(task_1)
+        task_2 = simulator.run(...)
+        project.add_task(task_2)
     """
 
-    def __init__(self, name: str, *, append: bool = False):
+    def __init__(self, name: str):
         """Initialize the Project instance.
-        
-        Args:
-          name (str): The name of the project.
-          append (bool): A flag indicating that new tasks can be appended to
-              the project. If set to False (default), task submission will fail,
-              even though the project might be opened.
-        """
-        self.append = append
-        self._token = None
-        self._list_of_tasks = None
-
-        model = self._get_model(name)
-        if model is None:
-            model = self._create_model(name)
-        self._update_from_api_response(model)
-
-    @classmethod
-    def _from_api_response(cls, model: ProjectModel) -> "Project":
-        """Builds a `Project` object from an API response.
 
         Args:
-          model: Info about the project. This should be fetched from the
-          backend.
-
-        """
-        project = cls.__new__(Project)
-        project.append = False
-        project._token = None
-        return project._update_from_api_response(model)
-
-    @staticmethod
-    def _get_model(name: str) -> ProjectModel:
-        """Fetches the project info from the backend.
-
-        Args:
-          name: The name of the project
-
-        Returns:
-          Project models object with the info about the project.
-
+          name (str): The name of the project.
         """
+        # If the project already exists, we will load it from the backend.
+        self._proj_data = self._get_project(name)
+        # Else, we will create a new project.
+        if not self._proj_data:
+            self._proj_data = self._create_project(name)
+
+    def _get_project(self, name: str):
+        """Fetches the project info from the backend."""
         try:
-            _logger.debug("Trying to get remote project %s", name)
-            api = projects_api.ProjectsApi(inductiva.api.get_client())
+            api = projects_api.ProjectsApi(inductiva_api.get_client())
```

Comment:
In order to not repeate this so many times, we could define the api in the __init__:
`self._api = projects_api.ProjectsApi(inductiva_api.get_client())`
```suggestion
    response = self._api.get_project({"name": name})
```

Reply:
Yes, thanks! Indeed, it would be particularly bad on the "add_task" method, because these can be called thousands of times if you are generating a dataset, for example...